Xwt.Gtk/Xwt.GtkBackend/BoxBackend.cs

```diff
@@ -94,14 +94,20 @@ public void ReplaceChild (Gtk.Widget oldWidget, Gtk.Widget newWidget)
 			Add (newWidget);
 			children [newWidget] = r;
 		}
-		
+
+		void UpdateFocusChain (Orientation orientation)
+		{
+			FocusChain = children.OrderBy ((arg) => orientation == Orientation.Horizontal ? arg.Value.Rect.X : arg.Value.Rect.Y).Select (arg => arg.Key).ToArray ();
```

Comment:
Don't like the use of LINQ here, would prefer creating a new array and using an EqualityComparer which looks into the dictionary for the WidgetData.